tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -0,0 +1,330 @@
+    Download a file from a URL. Modified from 
+    https://stackoverflow.com/questions/9419162/download-returned-zip-file-from-url
+    '''
+    r = requests.get(url, stream=True)
+    if r.status_code==404:
+        return ''
+    else:
+        if verbose:
+            print('Beginning download of {} to {}'.format(url, save_path))
+        with open(save_path, 'wb') as fd:
+            for chunk in r.iter_content(chunk_size=chunk_size):
+                fd.write(chunk)
+        if verbose:
+            print('Completed download of {} to {}'.format(url, save_path))
+        return save_path
+
+
+def readStationFile(filename):
+    '''
+    Read a list of GNSS station names from a plain text file
+    '''
+    statList = []
+    with open(filename, 'r') as f:
+        for line in f:
+            statList.append(line.strip())
+    return statList 
+
+
+def writeStationList(statList, filename = 'gnssStationList_overbbox.txt'):
+    '''
+    Write a python list of GNSS station names to a simple text file
+    '''
+    with open(filename, 'w') as f:
+        for stat in statList:
+            f.write('{}\n'.format(stat))
+
+
+def getStationList(bbox = None, writeLoc = None, userstatList = None):
+    '''
+    Creates a list of stations inside a lat/lon bounding box from a source
+    Inputs: 
+        bbox    - length-4 list of floats that describes a bounding box. Format is 
+                  S N W E
+    '''
+    if writeLoc is None:
+        writeLoc = os.path.join(os.getcwd(), 'gnssStationList_overbbox.csv')
+    else:
+        writeLoc = os.path.join(writeLoc, 'gnssStationList_overbbox.csv')
+
+    if userstatList:
+        with open(userstatList) as f:
+            userstatList = [line.rstrip('\n') for line in f]
+
+    statList = getStatsByllh(llhBox = bbox, userstatList = userstatList)
+
+    # write to file and pass final stations list
+    statList.to_csv(writeLoc, index=False)
+    stations = list(statList['ID'].values)
+
+    return stations, writeLoc
+
+
+def inBox(lat, lon, llhbox):
+    '''
+    Checks whether the given lat, lon pair are inside the bounding box llhbox
+    '''
+    lattest = lat < llhbox[1] and lat > llhbox[0]    
+    lontest = lon < llhbox[3] and lon > llhbox[2]
+    if lattest and lontest:
+        return True
+    else:
+        return False
+
+
+def fix_lons(lon):
+    """
+    Fix the given longitudes into the range ``[-180, 180]``.
+
+    """
+    fixed_lon = ((lon + 180) % 360) - 180
+    # Make the positive 180s positive again.
+    if fixed_lon == -180 and lon > 0:
+        fixed_lon *= -1
+
+    return fixed_lon
+
+
+def getID(line):
+    '''
+    Pulls the station ID, lat, and lon for a given entry in the UNR text file
+    '''
+    statID = str(line.split()[0].decode('utf-8'))
+    lat = float(line.split()[1].decode('utf-8'))
+    lon = float(line.split()[2].decode('utf-8'))
+
+    return statID, lat, lon
+
+
+def downloadTropoDelays(stats, years, writeDir = '.', download = False, verbose = False):
```

Comment:
Hey @sssangha can you modify the codes so that the main functions are at the top of the file, with helpers such as getID, fix_lons, etc. toward the bottom? Something like parseCMD at the top, followed by downloadTropoDelays, followed by download_UNR, etc. This way these scripts will be consistent with the updates from @Askaholic.

Reply:
True, restructured accordingly.